packages/runtime/src/ctx.ts

```diff
@@ -154,6 +155,11 @@ export class Ctx implements Context {
     toAccount.amount += BigInt(txParam.amountMicroAlgos); // add 'x' algo to receiver
     this.assertAccBalAboveMin(fromAccount.address);
 
+    if (txParam.payFlags.rekeyTo) {
+      const spend = txParam.payFlags.rekeyTo;
+      fromAccount.rekeyTo(spend);
+    }
```

Comment:
so we can rekey only for `TransferAlgo` transaction? I think we can rekey account with more txtypes (asset transfer, call app, optin ..etc) as well. Could you please check/confirm?

Reply:
In Rekey document they wrote, "A rekey-to transaction is a `payment` type transaction which includes the rekey-to parameter set to a well-formed Algorand address." and in this [docs](https://developer.algorand.org/docs/get-details/transactions/transactions/#payment-transaction) we can see only ALGO transfer is a payment transaction.

Reply:
Could you please test it? I think it's for other types as well (otherwise we would need to update `algob` code as well). Maybe test with `axfer`?